### PR TITLE
walletrpc: add `startheight` parameter to the `ListSweeps` RPC

### PR DESCRIPTION
**Pull request by bhandras**
## Change Description
By default `ListSweeps` will list all sweeps known to the wallet. Over time this may become expensive to call and callers may not be interested in the full history. This commit adds a `startheight` parameter to the `ListSweeps` RPC call. This parameter can be used to fetch sweeps only from a specified start block height or only unconfirmed ones if it is set to -1.

## Steps to Test
Steps for reviewers to follow to test the change.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [x] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [x] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [x]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

### CODE REVIEW

File: cmd/lncli/walletrpc_active.go [hunk 1]
```diff
@@ -412,6 +412,14 @@ var listSweepsCommand = cli.Command{
 			Name:  "verbose",
 			Usage: "lookup full transaction",
 		},
+		cli.IntFlag{
+			Name: "startheight",
+			Usage: `The start height to use when fetching sweeps.
```
Comment (yyforyongyu) [inline cmd/lncli/walletrpc_active.go:417]: Let's write an itest to validate the description here? I'm curious about the `-1` value here as we already use `btcwallet.UnconfirmedHeight` as our end height so it means the unconfirmed txes are always included?

**Review by yyforyongyu — CHANGES_REQUESTED** [review]
Very nice addition! Think we just miss an itest to check the behavior.

File: cmd/lncli/walletrpc_active.go [hunk 2]
```diff
@@ -412,6 +412,14 @@ var listSweepsCommand = cli.Command{
...
+                                If not specified (0), the result will start from
```
Comment (guggero) [inline cmd/lncli/walletrpc_active.go:418]: nit: we normally don't use back tick strings for the flag descriptions, as those will include the tab characters used for indenting. That then looks weird on narrow terminals (where the CLI library will attempt to render the flag text according to the terminal width).

File: cmd/lncli/walletrpc_active.go (hunk 1 above)
Comment (bhandras) [reply cmd/lncli/walletrpc_active.go]: done

File: itest/lnd_onchain_test.go [hunk 3]
```diff
@@ -832,3 +832,100 @@ func testRemoveTx(ht *lntest.HarnessTest) {
 	expectedAmt = btcutil.Amount(initialWalletAmt) - txFee
 	require.EqualValues(ht, expectedAmt, aliceBalResp.ConfirmedBalance)
 }
+
+// testListSweeps tests that we are able to:
+// - list only those sweeps that are currently in the mempool,
+// - list sweeps given a starting block height,
+// - list all sweeps.
+func testListSweeps(ht *lntest.HarnessTest) {
```
Comment (yyforyongyu) [inline itest/lnd_onchain_test.go:840]: 🙏

**Review by yyforyongyu — APPROVED** [review]
LGTM 🌷

**Pull request merged**